### PR TITLE
add ratelimit configmap

### PR DESCRIPTION
**Pull request by zackzhangkai**
Signed-off-by: zackzhangkai <zhangkaiamm@gmail.com>

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

should add configmap to this file, otherwise, the pod cannot run well.

### CODE REVIEW

**Review by bianpengyuan — COMMENTED** [review]
This yaml file is meant for only deploying the rate limit service, and the guide (which references this file) has a step to configure the config map before deploying the workload: https://preliminary.istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#global-rate-limit So I think we should not have a config map defined here...

**Comment by bianpengyuan** [comment]
As an alternative, can we add a warning (`a rate limit configmap is needed to make the rate limit deployment run properly`) and an example configmap like what you added to the comment at the beginning of this file?

**Comment by zackzhangkai** [comment]
> As an alternative, can we add a warning (`a rate limit configmap is needed to make the deployment run properly`) and an example configmap like what you added to the comment at the beginning of this file? Or simply make the volume optional, as long as the rate limit service has proper file watch it should work fine.

@bianpengyuan configmap in the comment as you said seems good

File: samples/ratelimit/rate-limit-service.yaml [hunk 1]
```diff
@@ -15,6 +15,26 @@
 ##################################################################################################
 # Redis service and deployment
 # Ratelimit service and deployment
+
+# Warning: configmap is required as follows:
```
Comment (bianpengyuan) [inline samples/ratelimit/rate-limit-service.yaml:19]:
```suggestion
# Note: a configmap is needed to make the rate limit deployment work properly, for example:
```

**Review by bianpengyuan — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**